### ISSUE TITLE
cox.zph() implementation and evaluations

### ISSUE THREAD

**Issue by jintaili** [therneau/survival#97]
Hi! I am trying to use `cox.zph()` to assess proportionality assumptions in my data set. However, I am having trouble 1) reproducing some examples and 2) interpret results on my own data set. Specifically:

1) I tried to reproduce examples from the documentation and [here](http://www.sthda.com/english/wiki/cox-model-assumptions#infos). In the latter, in my output of the cox.zph() call, there was not `rho` column but `df` instead. 
In addition, while I could use the plot.cox.zph() function, when I called ggcoxzph(), the plot of every covariate shared the same standard error, with no fitted line.

2) When I tried to apply the same methods on my own data set, due to the sheer number of observations, the plot of plot.cox.zph() got flooded with points and became difficult to read. But with ggcoxzph(), again every covariate share the same `se` with no fitted spline in the output.
Moreover, I am trying to find what the points stand for in the plots and any reference would be great. Thanks!

survival version: 3.1.8
survminer version: 0.4.6
sessionInfo():
```
R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.2

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] graphics  grDevices datasets  stats     utils     methods   base

other attached packages:
 [1] survminer_0.4.6                 ggpubr_0.2.4                    caret_6.0-84                    lattice_0.20-38
 [5] survAUC_1.0-5                   flexsurv_1.1.1                  survival_3.1-8                  plyr_1.8.4
 [9] plotly_4.9.1                    ggplot2_3.2.1                   bigrquery_1.2.0                 jsonlite_1.6
[13] dtplyr_0.0.3                    rsutils_6.07.00010110248        rsugeneral_6.07.000485049       rsuxls_6.07.000495052
[17] rsuworkspace_6.07.000539753     rsuvydia_6.07.000995099         rsushiny_6.07.00010110057       rsuscrubbers_1.09.003545398
[21] rsuprophesize_1.09.003545052    rsuplotting_6.07.0001029751     rsunotify_6.07.000575751        rsujesus_6.07.0005551101
[25] rsuaws_6.07.000985053           rsudb_6.07.000495050            rsucurl_1.09.003535750          rsuconsoleutils_6.07.0005110250
[29] rsubitly_1.09.0035650101        rsuaspath_6.07.000564951        rcreds_0.7.7                    collectArgs_0.5.0
[33] magrittr_1.5                    dplyr_0.8.3                     data.table_1.12.2               colorout_1.2-2

loaded via a namespace (and not attached):
 [1] colorspace_1.4-1     ggsignif_0.6.0       class_7.3-15         rstan_2.19.2         remotes_2.1.0        ggrepel_0.8.1
 [7] bit64_0.9-7          prodlim_2019.10.13   mvtnorm_1.0-11       lubridate_1.7.4      codetools_0.2-16     splines_3.6.1
[13] knitr_1.25           zeallot_0.1.0        km.ci_0.5-2          broom_0.5.2          shiny_1.4.0          compiler_3.6.1
[19] httr_1.4.1           backports_1.1.5      assertthat_0.2.1     Matrix_1.2-17        fastmap_1.0.1        lazyeval_0.2.2
[25] cli_1.1.0            later_1.0.0          formatR_1.7          htmltools_0.4.0      prettyunits_1.0.2    tools_3.6.1
[31] gtable_0.3.0         glue_1.3.1           reshape2_1.4.3       Rcpp_1.0.2           prophet_0.5          vctrs_0.2.0
[37] nlme_3.1-140         iterators_1.0.12     timeDate_3043.102    xfun_0.10            gower_0.2.1          stringr_1.4.0
[43] ps_1.3.0             mime_0.7             lifecycle_0.1.0      muhaz_1.2.6.1        zoo_1.8-6            MASS_7.3-51.4
[49] scales_1.0.0         microbenchmark_1.4-7 ipred_0.9-9          promises_1.1.0       parallel_3.6.1       inline_0.3.15
[55] RColorBrewer_1.1-2   gridExtra_2.3        KMsurv_0.1-5         loo_2.1.0            StanHeaders_2.19.0   rpart_4.1-15
[61] stringi_1.4.3        foreach_1.4.7        pkgbuild_1.0.6       lava_1.6.6           rlang_0.4.0          pkgconfig_2.0.3
[67] matrixStats_0.55.0   bitops_1.0-6         purrr_0.3.3          recipes_0.1.7        htmlwidgets_1.5.1    labeling_0.3
[73] bit_1.1-14           processx_3.4.1       tidyselect_0.2.5     deSolve_1.27.1       R6_2.4.0             generics_0.0.2
[79] DBI_1.0.0            pillar_1.4.2         withr_2.1.2          RCurl_1.95-4.12      nnet_7.3-12          tibble_2.1.3
[85] mstate_0.2.12        crayon_1.3.4         survMisc_0.5.5       grid_3.6.1           callr_3.3.2          ModelMetrics_1.2.2
[91] digest_0.6.21        xtable_1.8-4         tidyr_1.0.0          httpuv_1.5.2         stats4_3.6.1         munsell_0.5.0
```

**Comment by Vivek-Prakash-US** [comment]
They are using an older version of the survival package in that tutorial. You should be able to get that version with:

install.packages('versions')
versions::install.versions('survival', '2.44-1.1')

**Comment by therneau** [comment]
The cox.zph function has been updated in survival 3 in order to deal with multi-state models.   The rho parameter in the printout was removed as part of that.   (I no longer believe it was ever a useful column).   The formal test was tightened up, i.e., the code used to have an approximation in one spot to save computation time, which is now replaced with the correct computation.  (The approximation was usually okay.)

 For plot.cox.zph, use the resid=FALSE option to remove the points.   For small data sets it makes sense to look at whether the plot has been unduly influenced by an outlier, but for larger data sets you are correct that the residuals simply flood the page.   I myself use residuals, rug, or nothing for small, medium, and large data sets.  I use rug(zpfit$x) as a separate call, but I really should add the argument as an option.

I don't personally use ggcoxzph.  I agree that the plots are not correct.   I ran all of the test suites for survminer before submitting survival version 3, but changes in a graph are one area that test suites don't cover.   I have cc'd this note to the maintainer of survminer.
